extensions/markdown/src/extension.ts

```diff
@@ -0,0 +1,170 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as vscode from 'vscode';
+import * as path from 'path';
+import { ExtensionContext, TextDocumentContentProvider, EventEmitter, Event, Uri, ViewColumn } from "vscode";
+
+const hljs = require('highlight.js');
+const mdnh = require('markdown-it-named-headers');
+const md = require('markdown-it')({
+	html: true,
+	highlight: function (str, lang) {
+		if (lang && hljs.getLanguage(lang)) {
+			try {
+				return `<pre class="hljs"><code><div>${hljs.highlight(lang, str, true).value}</div></code></pre>`;
+			} catch (error) { }
+		}
+		return `<pre class="hljs"><code><div>${md.utils.escapeHtml(str)}</div></code></pre>`;
+	}
+}).use(mdnh, {});
```

Comment:
Sounds great.  Could you help me try to understand a little better how to implement this?  I believe that `.use` is on the 'on-the-fly' as in it should work something like this:

``` javascript
var md = require('markdown-it')();
var mathPlugin = require('markdown-it-katex');

md.render('$1 \\over 2$'); // renders '$1 \\over 2$' in paragraph tag

md = md.use(mathPlugin);
md.render('$1 \\over 2$'); // renders '1/2' in new fancy math
```

So you could certainly iterate over a list of plugins to attach them.  The issue is that I am not exactly sure how I would get the extension to obtain the `mathPlugin` object from a settings list.  Also how to make this list persist through extension updates.  Do you have any ideas?

Minor note:  There are some other methods for extending `markdown-it` where you override the rendering functions to meet special needs.  It's very similar to [what vscode already has](https://github.com/Microsoft/vscode/blob/65cb3397673b922c1b6759d145a3a183feb3ee5d/src/vs/languages/markdown/common/markdownWorker.ts#L132).  But just getting the plugins to load would be great and worry about this later maybe.